File: sea_battle.py
n = int(input())
rule = {1: 4, 2: 3, 3: 2, 4: 1}
for _ in range(n):
    ships = map(int, input().split())
    ships_map = {}
    for ship in ships:
        ships_map[ship] = ships_map.get(ship, 0)+1
    if ships_map == rule:
        print("YES")
    else:
        print("NO")

File: sea_battle.go
package main

import "fmt"

func main() {
    var n int
    fmt.Scan(&n)
    rule := map[int]int{1: 4, 2: 3, 3: 2, 4: 1}
    for i := 0; i < n; i++ {
        test := map[int]int{1: 0, 2: 0, 3: 0, 4: 0}
        for j := 0; j < 10; j++ {
            var ship int
            fmt.Scan(&ship)
            test[ship]++
        }
        if equalMaps(rule, test) {
            fmt.Println("YES")
        } else {
            fmt.Println("NO")
        }
    }
}

func equalMaps(m1, m2 map[int]int) bool {
    if len(m1) != len(m2) {
        return false
    }
    for k, v := range m1 {
        if m2[k] != v {
            return false
        }
    }
    return true
}